### PR TITLE
Add documentation about access denied handler

### LINKED ISSUE #5139: [Security] Missing documentation regarding AccessDeniedHandlerInterface
I've seen a few occasion where custom access denied handling is required. It's not rocket science to implement, you put `access_denied_handler: service.id` on your firewall configuration and implement the `AccessDeniedHandlerInterface` in this service. 

I think this could be a simple explanation but add some real value to the docs.

### PR DESCRIPTION
**Pull request by nykopol**
Fix #5139

### CODE REVIEW

File: security/access_denied_handler.rst [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+Then you must register your service :
+
+.. code-block:: yaml
+
+    # app/config/services.yml
+    services:
+        custom_handler.service.id:
+            class: AppBundle\Security\AccessDeniedHandler
+
+That's it, now on the ``foo`` firewall, all ``AccessDeniedException`` will be notified to you service.
```
Comment (ogizanagi) [inline security/access_denied_handler.rst]:
```diff
- to you service.
+ to your service.
```

File: security/access_denied_handler.rst [hunk 2]
```diff
@@ -0,0 +1,68 @@
+.. index::
+    single: Security; Creating a Custom Access Denied Handler
+
+How to Create a Custom Access Denied Handler
+============================================
+
+When your application throw an ``AccessDeniedException`` you can catch this exception
```
Comment (xabbuh) [inline security/access_denied_handler.rst]: throws

Comment (xabbuh) [inline security/access_denied_handler.rst]: missing comma before "you"

File: security/access_denied_handler.rst [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+        // app/config/security.php
+        $container->loadFromExtension('security', array(
+            'firewalls' => array(
+                'foo' => array(
+                    // ...
+                    'access_denied_handler' => 'custom_handler.service.id',
+                ),
+            ),
+        ));
```
Comment (xabbuh) [inline security/access_denied_handler.rst]: Can you please also add an XML config example?

File: security/access_denied_handler.rst [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+
+Your handler must implement the interface
```
Comment (xabbuh) [inline security/access_denied_handler.rst]: we can omit "interface" here

File: security/access_denied_handler.rst [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+:class:`Symfony\\Component\\Security\\Http\\Authorization\\AccessDeniedHandlerInterface`.
+This interface define one method called ``handle()`` that can do whatever you want.
```
Comment (xabbuh) [inline security/access_denied_handler.rst]: I would reword this a bit:

> [...] that implements the logic you want to execute when access is denied to the current user.

File: security/access_denied_handler.rst [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+You can use it to send a mail, log a message, or generally return a custom Response.
```
Comment (xabbuh) [inline security/access_denied_handler.rst]: This could then be enclosed with parentheses after the previous sentence.

File: security/access_denied_handler.rst [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+
```
Comment (xabbuh) [inline security/access_denied_handler.rst]: Please remove one blank line.

File: security/access_denied_handler.rst [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+.. code-block:: php
+
+    namespace AppBundle\Security;
```
Comment (xabbuh) [inline security/access_denied_handler.rst]: We should better not use AppBundle in this context.

File: security/access_denied_handler.rst [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+
+    use Symfony\Component\Security\Http\Authorization\AccessDeniedHandlerInterface;
+    use Symfony\Component\HttpFoundation\Request;
+    use Symfony\Component\HttpFoundation\Response;
+    use Symfony\Component\Security\Core\Exception\AccessDeniedException;
```
Comment (xabbuh) [inline security/access_denied_handler.rst]: Can you please sort the `use` statements alphabetically?

File: security/access_denied_handler.rst (hunk 8 above)
Comment (nykopol) [inline security/access_denied_handler.rst]: What would be a good place for this class ? For me `AppBundle\Security` look like a perfect default place for a `Foo` project.

**Comment by nykopol** [comment]
@xabbuh thanks for your review.

**Review by javiereguiluz — APPROVED** [review]
👍 another missing feature documented! Thanks a lot @nykopol.

I've made some minor rewords in your original article. If you don't agree with something, please tell us. Thanks!

**Comment by nykopol** [comment]
@javiereguiluz LGTM, thanks

**Comment by xabbuh** [comment]
Thank you @nykopol for your great work on this PR. I have merged your changes and made some minor tweaks in c69805154b9a463f6f97982ef5fdd47db4e1bab7.
